File: bank-account-app-back/exalt-hexagonal-archi-kafka-keycloak-bs-ms-operation-api/src/main/java/com/exalt/exalthexagonalarchikafkakeycloakbsmsoperationapi/infrastructure/adapters/output/dtos/TransferResponseDto.java
package com.exalt.exalthexagonalarchikafkakeycloakbsmsoperationapi.infrastructure.adapters.output.dtos;

import com.exalt.exalthexagonalarchikafkakeycloakbsmsoperationapi.infrastructure.adapters.feign_clients.domain.AccountResponseDto;

import java.math.BigDecimal;
import java.time.Instant;

public record TransferResponseDto(String transferId, BigDecimal transferAmount, String description, Instant createdAt,AccountResponseDto origin, AccountResponseDto destination) {
}

File: bank-account-app-front/src/app/shared/models/backend/operation/transfer-response.dto.ts
import { AccountResponseDto } from "../account/account-response.dto";

export interface TransferResponseDto {
    transferId: string;
    transferAmount: number;
    description: string;
    createdAt: Date;
    originAccount: string;
    destinationAccount: string;
    origin: AccountResponseDto;
    destination: AccountResponseDto;
}